xarray/core/dataset.py

```diff
@@ -3393,11 +3399,20 @@ def roll(self, **shifts):
         if invalid:
             raise ValueError("dimensions %r do not exist" % invalid)
 
+        if roll_coords is None:
+            warnings.warn("roll_coords will be set to False in the future."
+                          " Explicitly set roll_coords to silence warning.",
+                          DeprecationWarning, stacklevel=3)
```

Comment:
Also -- consider using `stacklevel=2` here. `stacklevel=3` here would be confusing if `Dataset.roll()` is called directly.

I think it's *probably* better to incorrect indicate that the error is being raised internally inside xarray rather than to incorrectly indicate that it's being raised by some arbitrary user code.